pkg/strategy/controller_test.go

```diff
@@ -20,34 +17,24 @@ func TestNewControllerForAllInOneAsDefault(t *testing.T) {
 	jaeger := v1alpha1.NewJaeger("TestNewControllerForAllInOneAsDefault")
 
 	ctrl := For(context.TODO(), jaeger)
-	rightType := false
-	switch ctrl.(type) {
-	case *allInOneStrategy:
-		rightType = true
-	}
-	assert.True(t, rightType)
+	assert.Equal(t, ctrl.Type(), AllInOne)
 }
 
 func TestNewControllerForAllInOneAsExplicitValue(t *testing.T) {
 	jaeger := v1alpha1.NewJaeger("TestNewControllerForAllInOneAsExplicitValue")
 	jaeger.Spec.Strategy = "ALL-IN-ONE" // same as 'all-in-one'
 
 	ctrl := For(context.TODO(), jaeger)
-	rightType := false
-	switch ctrl.(type) {
-	case *allInOneStrategy:
-		rightType = true
-	}
-	assert.True(t, rightType)
+	assert.Equal(t, ctrl.Type(), AllInOne)
 }
 
 func TestNewControllerForProduction(t *testing.T) {
 	jaeger := v1alpha1.NewJaeger("TestNewControllerForProduction")
 	jaeger.Spec.Strategy = "production"
+	jaeger.Spec.Storage.Type = "elasticsearch"
 
 	ctrl := For(context.TODO(), jaeger)
-	ds := ctrl.Create()
-	assert.Len(t, ds, 6)
+	assert.Len(t, ctrl.Deployments(), 2)
```

Comment:
Just wanted to check why the number of deployments has changed from 6 to 2?

Reply:
Before, `Create()`  returned all the objects that should be created, like deployments, daemonsets, configmaps, and so on. `Deployments()` return only the deployments. `ds` was probably a bad name for that var.

Reply:
By the way: I have not changed the test to check all the objects, as it's done already within the test for the concrete strategy. Instead, I just needed a quick way to ensure the code returns the right strategy. As we now have a `Type()` method, I changed the test to assert the strategy type instead.